crates/flipperzero/src/furi/log/mod.rs

```diff
@@ -0,0 +1,166 @@
+//! Furi Logging system.
+
+mod metadata;
+pub use metadata::{Level, LevelFilter};
+
+/// The standard logging macro.
+///
+/// This macro will generically log with the specified `Level` and C-style format argument
+/// list.
+///
+/// # Examples
+///
+/// ```
+/// use flipperzero::{error, furi::log::Level};
+///
+/// # fn main() {
+/// let error_code = 42;
+/// log!(Level::ERROR, "Failed to handle the florp: %d", error_code);
+/// log!(target: "events", Level::INFO, "Finished the documentation!");
+/// # }
+/// ```
+#[macro_export]
+macro_rules! log {
+    (target: $target:expr, $lvl:expr, $msg:expr $(, $arg:expr)*) => ({
+        if $lvl <= $crate::furi::log::LevelFilter::current() {
+            unsafe {
+                $crate::__macro_support::__sys::furi_log_print_format(
+                    $crate::__macro_support::__level_to_furi($lvl),
+                    $crate::__macro_support::__sys::c_string!($target),
+                    $crate::__macro_support::__sys::c_string!($msg),
+                    $($arg,)*
```

Comment:
Currently we just pass in whatever arguments the caller provides. I've confirmed that this does work, but if we spent some time on macro magic we could probably type-check the arguments against the format parameters in `$msg`.

Reply:
Hmm, a downside to using `ufmt` is that it means logging requires `alloc`, because we need a `String` to accumulate the output into before calling `sys::furi_log_print_format`. Though we could technically avoid enabling `alloc` on the Rust side by using `FuriString` instead? I can implement a Rust wrapper around that.

Reply:
Started this in #50.

Reply:
I have this working locally.